### DISCUSSION TITLE
Class "Str" not found

### DISCUSSION THREAD

**Discussion by eminos** [filamentphp/filament#227] [Help]
**Describe the bug**

Class "Str" not found (View: /app/vendor/filament/filament/packages/forms/resources/views/components/field-group.blade.php)

A new Filament installation on an old/upgraded Laravel app.
It's Laravel 8.0 but it has been upgraded for a few years, and I'm suspecting I've missed some Laravel upgrade steps regarding the String helpers.

Anyway, it might be something you might want to fix.

**Comment by eminos** [comment]
Adding `'Str' => Illuminate\Support\Str::class,` to `config/app.php` aliases array, solved this.

**Comment by danharrin** [comment]
Hey, can you please ensure that the alias is registered in your config like this?

https://stackoverflow.com/a/59503877

**Reply by viezel** [reply]
@danharrin you should always use fully qualified namespace or import in use statements. Please don’t rely on class aliases as it degrades type safety.

**Reply by ryangjchandler** [reply]
> @danharrin you should always use fully qualified namespace or import in use statements. Please don’t rely on class aliases as it degrades type safety.

This is inside of a Blade view, so imports won't work. I generally agree though, using the FQN is much safer. Saying that, it's not our fault if a Laravel application is misconfigured.

**Reply by danharrin** [reply]
Can pivot to FQN if you wish? Just a find and replace job.

**Reply by viezel** [reply]
Its not a huge deal but there is a shift in the Laravel community to more type safety. Tools like Rector will automatically remove class aliases and psalm will report them as errors. 
I do not consider removing class aliases a misconfiguration.

**Reply by viezel** [reply]
here is the link: https://getrector.org/blog/2020/12/07/laravel-facades-to-constructor-injection-replace-facade-aliases-with-full-classes-in-2-hours

**Comment by ryangjchandler** [comment]
@danharrin Think it's safe to close this, user error.

**Comment by danharrin** [comment]
Sure, I'll move it to a discussion.
